packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -326,23 +333,26 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
         bytes32 depositHash = _getDepositHash(_depositData);
         RelayData storage relay = relays[depositHash];
 
-        // Note `hasPrice` will return false if liveness has not been passed in the optimistic oracle.
-        require(
-            relays[depositHash].relayState == RelayState.Pending &&
-                _getOptimisticOracle().hasPrice(
```

Comment:
I'm not convinced of this technique after thinking more about it. What happens if I call `settle` directly on the OO first (as soon as the request is passed liveness) before the relayer calls `settleRelay`? In this case, the price request will be settled _before_ entering this method making the subsequent call to the OO on `settle` revert. This will make this method revert.

As a result, the path forward is: a) calling `settle` if the request is settleable or using `hasPrice` b) leaving it as it was before wherein we ignore the `settle` logic thereby enforcing that it is settled outside of this contract.

Reply:
This is a good point. We can wrap try catch around settle and then re add the hasPrice requirement?

Reply:
I ended up using the `settleAndGetPrice` which does not revert if the price is already settled and returns the resolved price (after optionally settling).

Reply:
excellent. I think this is the ideal setup. Also really happy you are checking the value price to remove any complexity in _implied_ correctness.